clients/src/main/java/org/apache/kafka/clients/consumer/StickyAssignor.java

```diff
@@ -422,10 +503,18 @@ private int getBalanceScore(Map<String, List<TopicPartition>> assignment) {
             TreeSet<String> sortedConsumers = new TreeSet<>(new SubscriptionComparator(assignments));
             sortedConsumers.addAll(assignments.keySet());
 
+            List<TopicPartition> prevAssignedPartitions = new ArrayList<>(prevAssignment.keySet());
             while (!sortedConsumers.isEmpty()) {
                 String consumer = sortedConsumers.pollLast();
                 List<TopicPartition> remainingPartitions = assignments.get(consumer);
-                if (!remainingPartitions.isEmpty()) {
+                List<TopicPartition> prevPartitions = new ArrayList<>(prevAssignedPartitions);
```

Comment:
Do we need `prevAssignedPartitions`? Could we use `prevAssignment.keySet()` here? Some of these operations could probably be more efficient by working sets, but it may not matter too much since this is not a common case.

I'm having a little trouble following the logic here. I find it a little puzzling that we only consider the key set of `prevAssignment`, but I am probably missing something. Perhaps we can add some comments?